src/Sarif.Driver/Sdk/MultithreadedAnalyzeCommandBase.cs

```diff
@@ -319,7 +319,7 @@ public virtual TContext InitializeGlobalContextFromOptions(TOptions options, ref
                 context.TargetFileSpecifiers = options.TargetFileSpecifiers?.Any() == true ? InitializeStringSet(options.TargetFileSpecifiers) : context.TargetFileSpecifiers;
                 context.InvocationPropertiesToLog = options.InvocationPropertiesToLog?.Any() == true ? InitializeStringSet(options.InvocationPropertiesToLog) : context.InvocationPropertiesToLog;
                 context.Traces = options.Trace.Any() ? InitializeStringSet(options.Trace) : context.Traces;
-                context.RuleKinds = options.RuleKinds;
+                context.RuleKinds = options.RuleKindOption != null ? options.RuleKinds : context.RuleKinds;
```

Comment:
>options.RuleKindOption != null ? options.RuleKinds : context.RuleKinds [](http://example.com/codeflow?start=36&length=70)

Can we use this simpler syntax?

options.RuleKindOption ?? context.RuleKinds

Reply:
no we can not,
options.RuleKindOption is not the same type as options.RuleKinds

Reply:
if you see lines above
`context.ResultKinds = options.Kind != null ? options.ResultKinds : context.ResultKinds;`
Some tricky here, the `options.Kind` is not the same type as `options.ResultKinds`.

p.s. I think the naming before is not good for `Kind` for `ResultKinds` , it is better to name using xxxOption, or some pattern. People will not know `Kind` is the command line option for `ResultKinds`